src/SectorView.cpp

```diff
@@ -43,6 +43,236 @@ static const float FAR_THRESHOLD = 7.5f;
 static const float FAR_LIMIT = 36.f;
 static const float FAR_MAX = 46.f;
 
+class SectorView::Label {
+public:
+	// check if the cursor has hit the label, and if so, also save the boxes to select the text
+	// should run inside imgui context
+	virtual bool Hovered(float x, float y) = 0;
+	virtual void Draw(ImDrawList &dl, bool hovered = false) = 0;
+	virtual void OnClick() = 0;
+	float Depth() const { return pos.z; }
+	Label(SectorView &host, const vector3f &pos, ImU32 color) :
+		pos(pos), color(color), host(host){};
+
+protected:
+	vector3f pos;
+	ImU32 color;
+	SectorView &host;
+};
+
+// helper functions to detect when the cursor hits the label
+static bool inbox(float xmin, float xmax, float ymin, float ymax, float x, float y)
+{
+	return x > xmin && x < xmax && y > ymin && y < ymax;
+}
+
+static bool incircle(float cx, float cy, float r, float x, float y)
+{
+	return (x - cx) * (x - cx) + (y - cy) * (y - cy) < r * r;
+}
+
+static bool indiamond(float cx, float cy, float rad, float x, float y)
+{
+	float dx = x - cx;
+	float dy1 = y - (cy - rad);
+	float dy2 = y - (cy + rad);
+	return dy1 > dx && dy1 > -dx && dy2 < dx && dy2 < -dx;
+}
+
+class SectorView::StarLabel : public SectorView::Label {
+
+	std::string name;
+	std::array<float, 2> namePos;
+	// screen radius of the star
+	float radius;
+	// the height of the text of each label is different
+	float textHeight;
+	float scaledGap;
+	SystemPath path;
+
+	//  how much we will increase the original font so that it is less blurry for heavily enlarged labels
+	static constexpr float fontScale = 5.f;
+
+public:
+	StarLabel(SectorView &host, const vector3f &pos, const Color &color_, const std::string &name, const float radius, const SystemPath &path) :
+		Label(host, pos, 0),
+		name(name),
+		radius(radius),
+		path(path)
+	{
+		ImFont *&font = host.m_labels.fonts.starLabel;
```

Comment:
It's not necessary (and honestly just confusing) to take a reference to a pointer variable that you're using in read-only fashion; this should just be `ImFont *`.

Reply:
Done, where this pointer is not supposed to change